activerecord/lib/active_record/errors.rb

```diff
@@ -130,7 +130,8 @@ def initialize(message = nil, model = nil, primary_key = nil, id = nil)
 
   # Raised by {ActiveRecord::Base#save!}[rdoc-ref:Persistence#save!] and
   # {ActiveRecord::Base.create!}[rdoc-ref:Persistence::ClassMethods#create!]
-  # methods when a record is invalid and cannot be saved.
+  # If any of the <tt>before_*</tt> callbacks throws +:abort+, See
```

Comment:
This is not the only case where `RecordNotSaved` is raised; it's also raised when the record fails validation, for example.

Reply:
Thanks, I updated it, hopefully it's more clear now.